File: app/src/main/java/seveida/firetvforreddit/response/objects/Image.java
package seveida.firetvforreddit.response.objects;

import java.util.List;
import com.squareup.moshi.Json;

public class Image {

    @Json(name = "source")
    public Source source;
    @Json(name = "resolutions")
    public List<Resolution> resolutions = null;
    @Json(name = "variants")
    public Variants variants;
    @Json(name = "id")
    public String id;

    /**
     * No args constructor for use in serialization
     *
     */
    public Image() {
    }

    /**
     *
     * @param id
     * @param source
     * @param variants
     * @param resolutions
     */
    public Image(Source source, List<Resolution> resolutions, Variants variants, String id) {
        super();
        this.source = source;
        this.resolutions = resolutions;
        this.variants = variants;
        this.id = id;
    }

}

File: app/src/main/java/seveida/firetvforreddit/response/objects/Image.kt
package seveida.firetvforreddit.response.objects

import com.squareup.moshi.Json

class Image {

    @Json(name = "source")
    var source: Source
    @Json(name = "resolutions")
    var resolutions: List<Resolution>? = null
    @Json(name = "variants")
    var variants: Variants
    @Json(name = "id")
    var id: String

    /**
     * No args constructor for use in serialization
     *
     */
    constructor() {}

    /**
     *
     * @param id
     * @param source
     * @param variants
     * @param resolutions
     */
    constructor(source: Source, resolutions: List<Resolution>, variants: Variants, id: String) : super() {
        this.source = source
        this.resolutions = resolutions
        this.variants = variants
        this.id = id
    }

}
